block/internal/syncing/syncer.go

```diff
@@ -252,61 +267,94 @@ func (s *Syncer) processLoop() {
 }
 
 func (s *Syncer) startSyncWorkers() {
-	s.wg.Add(3)
+	s.wg.Add(2)
 	go s.daWorker()
-	go s.p2pWorker()
 	go s.pendingWorker()
 }
 
-func (s *Syncer) daWorker() {
-	defer s.wg.Done()
-	s.daWorkerLoop()
+func (s *Syncer) startP2PListeners() error {
+	if s.headerStore == nil {
```

Comment:
nit: `headerStore` and `dataStore` are used before in `Start()`. You can safely assume that they are not nil. The checks can go into constructor if you want them.